hydra-nl/process_hydranl.py

```diff
@@ -0,0 +1,127 @@
+ # -*- coding: utf-8 -*-
+"""
+Created on Thu Sep 1 12:00:00 2022
+
+This is a script to create process the hydra-nl calculations in a format suitable for KPZSS Illustratiepuntenmethode
+
+@author: BADD
+
+The following steps are done:
+1. Import modules
+2. Set paths
+3. Create empty dictionairy for results
+4. Loop over Hydra-NL folders with certain keywords
+5. Store input- and output variables from the Hydra-NL folders
+
+"""
+
+# Import modules
+from cmath import nan
+from operator import contains, index
+import string
+from reader import lees_invoerbestand, lees_uitvoerhtml, lees_illustratiepunten, lees_ofl
+from pathlib import Path
+import os
+import pandas as pd
+import numpy as np
+
+# set paths
+hydra_werkmap = r"C:\MyPrograms\Hydra-NL_KP_ZSS\werkmap"
+okader_norm_csv = pd.read_csv(r"D:\Users\BADD\Desktop\KP ZSS\kust\koppelingen\Pf_Vak_Eis_20220718.csv", delimiter=';')
+df_hydra_definition = pd.read_csv(r"D:\Users\BADD\Desktop\KP ZSS\GIS-WZ\okader_fc_hydra_unique_handedit_WZ_v2.csv")
+
+# create dictionairy to store results
+hydra_results = {
+    'OKADER VakId':[], 
+    'Uitvoerbestand':[], 
+    'Terugkeertijd':[], 
+    'q [m3/s/m]':[], 
+    'ZSS-scenario':[], 
+    'HYD_location_name':[], 
+    'HYD_location_X':[], 
+    'HYD_location_Y':[], 
+    'Profiel':[],
+    'HBN [m+NAP]':[], 
+    'Zeewaterstand [m+NAP]':[], 
+    'windsnelheid [m/s]':[], 
+    'h, teen [m+NAP]': [], 
+    'Hm0, teen [m]':[], 
+    'Tm-1.0, ts':[], 
+    'golfrichting [graden]':[], 
+    'windrichting [graden N]':[], 
+    'bijdrage ov. freq [%]':[]}
+
+
+for folder in os.listdir(hydra_werkmap):
+
+    # specify which keywords to match in hydra-nl werkmap
+    string_to_match = ["Waddenzee"]
+
+    for substring in string_to_match:
+        if substring in folder:
+
+            # find folders with string_to_match
+            for root, dirs, files in os.walk(os.path.join(hydra_werkmap, folder)):
+                    for file in files:
+
+                        # based on invoer.hyd the first columns of output dictionairy are filled
+                        if file.endswith('invoer.hyd'):
+                            location_invoer = os.path.join(root, file)
+                            invoer, terugkeertijden = lees_invoerbestand(location_invoer)
+                            
+                            # the entire path of profile is included in invoer.hyd, strip it to profile-name
+                            profielpath = invoer['algemeen']['PROFIEL']
+                            valueprofiel = profielpath.split("\\")[-1]
+                            valueprofiel = valueprofiel[:-1]
+                            value = valueprofiel.split(".")[0]
```

Comment:
Unclear what value is and now you're overwriting the variable which makes it hard to find out what happens to it along the way.

So give it a clear name and just perform these operations on valueprofiel in one line